[All Hosts] (sideloading) clarify sideloading on the web (#3620)
OfficeDev/office-js-docs-pr: docs/testing/sideload-office-add-ins-for-testing.md

The steps to sideload an add-in on the web vary based on these factors:

- The host application (for example, Excel, Word, Outlook).
- What tool created the add-in project (for example, Visual Studio, Yeoman generator for Office Add-ins, or neither)
- Whether you are sideloading to retail Office on the web or to a Microsoft 365 tenancy.

Go to the section or article that matches your scenario:

- If you are sideloading an Outlook Add-in, see the article [Sideload Outlook add-ins for testing](../outlook/sideload-outlook-add-ins-for-testing.md). *The remaining bullets apply to non-Outlook add-ins.*
- If the add-in was created with the [Yeoman generator for Office Add-ins](../develop/yeoman-generator-overview.md), see [Sideload a Yeoman-created add-in to Office on the web](#sideload-a-yeoman-created-add-in-to-office-on-the-web).
- If the add-in was created with Visual Studio, see [Sideload an add-in on the web when using Visual Studio](#sideload-an-add-in-on-the-web-when-using-visual-studio).
- For all other cases, see one of these sections:

>   - If you are sideloading to retail Office on the web, see [Sideload an add-in in Office on the web manually](#sideload-an-add-in-to-office-on-the-web-manually).

Reviewer: ```suggestion
  - If you are sideloading to retail Office on the web, see [Sideload an add-in to Office on the web manually](#sideload-an-add-in-to-office-on-the-web-manually).
```

>   - If you are sideloading to retail Office on the web, see [Manually sideload an add-in to Office on the web](#manually-sideload-an-add-in-to-office-on-the-web).

Reviewer: Is there a different option for VL?

Author: I don't understand the question. What is VL?

Reviewer: Sorry - I mean volume-licensed

Author: VL is desktop. There are other articles that cover sideloading on desktop. This aritcle is only about sideloading on the web.

Author: We clarified this off line. I changed the terminology to clarify that the distinction is between M365 account and Microsoft Account.

## Sideload a Yeoman-created add-in to Office on the web

This process is supported for **Excel**, **OneNote**, **PowerPoint**, and **Word** only. This example project assumes that you are using a project created with the [Yeoman generator for Office Add-ins](../develop/yeoman-generator-overview.md).

1. Open [Office on the web](https://office.live.com/) or OneDrive. Using the **Create** option, make a document in **Excel**, **OneNote**, **PowerPoint**, or **Word**. In this new document, select **Share** in the ribbon, select **Copy Link**, and copy the URL.

1. In the command line starting at the root directory of your project, run the following command. Replace "{url}" with the URL that you copied.

    [!INCLUDE [npm start:web command syntax](../includes/start-web-sideload-instructions.md)]

1. The first time you use this method to sideload an add-in on the web, you'll see a dialog asking you to enable developer mode. Select the checkbox for **Enable Developer Mode now** and select **OK**.

1. You will see a second dialog box, asking if you wish to register an Office Add-in manifest from your computer. You should select **Yes**.

> 1. Your add-in is installed. If it has an add-in command, it should appear on either the ribbon or the context menu. If it is a task pane add-in withour any add-in commands, the task pane should appear.

Reviewer: ```suggestion
1. Your add-in is installed. If it has an add-in command, it should appear on either the ribbon or the context menu. If it's a task pane add-in without any add-in commands, the task pane should appear.
```

## Sideload an add-in on the web when using Visual Studio

If you're using Visual Studio to develop your add-in, pressing F5 will automatically open an Office document in *desktop* Office, create a blank document, and sideload the add-in. When you want to sideload to *Office on the web*, the process to sideload is similar to manual sideloading to the web. The only difference is that you must update the value of the **SourceURL** element, and possibly other elements, in your manifest to include the full URL where the add-in is deployed.

1. In Visual Studio, show the **Properties** window by choosing **View** > **Properties Window**.
1. In the **Solution Explorer**, select the web project. This will display properties for the project in the **Properties** window.
1. In the Properties window, copy the **SSL URL**.

> 1. In the add-in project, open the manifest XML file. Be sure you are editing the source XML. For some project types Visual Studio will open a visual view of the XML which will not work for the next step.

Reviewer: ```suggestion
1. In the add-in project, open the manifest XML file. Be sure you are editing the source XML. For some project types, Visual Studio will open a visual view of the XML which won't work for the next step.
```

> 1. Search and replace all instances of **~remoteAppUrl/** with the SSL URL you just copied. You will see several replacements depending on the project type, and the new URLs will appear similar to `https://localhost:44300/Home.html`.

Reviewer: ```suggestion
1. Search and replace all instances of **~remoteAppUrl/** with the SSL URL you just copied. You'll see several replacements depending on the project type, and the new URLs will appear similar to `https://localhost:44300/Home.html`.
```

1. Save the XML file.

> 1. Right click the web project and choose **Debug** > **Start new instance**. This will run the web project without launching Office.

Reviewer: We should avoid using right click. Or you should provide an alternative instruction that works for other entry modes (e.g. keyboard). Linda Lu may have some suggestions.
```suggestion
1. Open the context menu of the web project (for example, by right clicking on it) then choose **Debug** > **Start new instance**. This runs the web project without launching Office.
```

> 1. From Office on the web, sideload the add-in using steps previously described in [Manually sideload an add-in to Office on the web manually](#manually-sideload-an-add-in-to-office-on-the-web).

Reviewer: ```suggestion
1. From Office on the web, sideload the add-in using steps previously described in [Manually sideload an add-in to Office on the web](#manually-sideload-an-add-in-to-office-on-the-web).
```

> ## Sideload an add-in to Office on the web manually

Reviewer: ```suggestion
## Manually sideload an add-in to Office on the web
```

> 1. From Office on the web, sideload the add-in using steps previously described in [Manually sideload an add-in to Office on the web](#manually-sideload-an-add-in-to-office-on-the-web).

Reviewer: You're referring to the section after this one, right?
```suggestion
1. From Office on the web, sideload the add-in using steps described in [Manually sideload an add-in to Office on the web](#manually-sideload-an-add-in-to-office-on-the-web).
```

This method doesn't use the command line and can be accomplished using commands only within the host application (such as Excel).

> 

Reviewer: Consider splitting out the first step so it's:
```
1. Open [Office on the web](https://office.com/). Open a document in **Excel**, **OneNote**, **PowerPoint**, or **Word**.
1. On the **Insert** tab on the ribbon in the **Add-ins** section, choose **Office Add-ins**.
```

Author: This suggestion, in effect, moves the first step of the "Manually ..." section to the end of the "... Visual Studio" section. The problem is that the "Manually ..." section is linked to from other articles and it has to have that first step in it.

---
title: Sideload Office Add-ins to Office on the web
description: Test your Office Add-in in Office on the web by sideloading.
ms.date: 09/02/2022
ms.localizationpriority: medium
---

# Sideload Office Add-ins to Office on the web

When you sideload an add-in, you're able to install the add-in without first putting it in an add-in catalog. This is useful when testing and developing your add-in because you can see how your add-in will appear and function.

When you sideload an add-in on the web, the add-in's manifest is stored in the browser's local storage, so if you clear the browser's cache, or switch to a different browser, you have to sideload the add-in again.

The steps to sideload an add-in on the web vary based on these factors:

- The host application (for example, Excel, Word, Outlook).
- What tool created the add-in project (for example, Visual Studio, Yeoman generator for Office Add-ins, or neither)

> - Whether you are sideloading to Office on the web with a Microsoft Account or with an account in a Microsoft 365 tenant.

Reviewer: @lindalu-MSFT I thought all the items in a list should end with a period if at least one item was more than 3 or so words long?

Go to the section or article that matches your scenario:

- If you are sideloading an Outlook Add-in, see the article [Sideload Outlook add-ins for testing](../outlook/sideload-outlook-add-ins-for-testing.md). *The remaining bullets apply to non-Outlook add-ins.*

> - If the add-in was created with the [Yeoman generator for Office Add-ins](../develop/yeoman-generator-overview.md), see [Sideload a Yeoman-created add-in to Office on the web](#sideload-a-yeoman-created-add-in-to-office-on-the-web).

Reviewer 2: ```suggestion
- If you created the add-in using the [Yeoman generator for Office Add-ins](../develop/yeoman-generator-overview.md), see [Sideload a Yeoman-created add-in to Office on the web](#sideload-a-yeoman-created-add-in-to-office-on-the-web).
```

> - If the add-in was created with Visual Studio, see [Sideload an add-in on the web when using Visual Studio](#sideload-an-add-in-on-the-web-when-using-visual-studio).

Reviewer 2: ```suggestion
- If you created the add-in using Visual Studio, see [Sideload an add-in on the web when using Visual Studio](#sideload-an-add-in-on-the-web-when-using-visual-studio).
```

> - For all other cases, see one of these sections:

Reviewer 2: ```suggestion
- For all other cases, see one of the following sections.
```

>   - If you are sideloading to Office on the web with a Microsoft Account, see [Manually sideload an add-in to Office on the web](#manually-sideload-an-add-in-to-office-on-the-web).

Reviewer 2: ```suggestion
  - If you're sideloading to Office on the web with a Microsoft Account, see [Manually sideload an add-in to Office on the web](#manually-sideload-an-add-in-to-office-on-the-web).
```

>   - If you are sideloading to Office on the web with an account in a Microsoft 365 tenant, see [Sideload an add-in to Microsoft 365](#sideload-an-add-in-to-microsoft-365).

Reviewer 2: ```suggestion
  - If you're sideloading to Office on the web with an account in a Microsoft 365 tenant, see [Sideload an add-in to Microsoft 365](#sideload-an-add-in-to-microsoft-365).
```

## Sideload a Yeoman-created add-in to Office on the web

> This process is supported for **Excel**, **OneNote**, **PowerPoint**, and **Word** only. This example project assumes that you are using a project created with the [Yeoman generator for Office Add-ins](../develop/yeoman-generator-overview.md).

Reviewer 2: ```suggestion
This process is supported for **Excel**, **OneNote**, **PowerPoint**, and **Word** only. This example project assumes you're using a project created with the [Yeoman generator for Office Add-ins](../develop/yeoman-generator-overview.md).
```

> 1. Open [Office on the web](https://office.live.com/) or OneDrive. Using the **Create** option, make a document in **Excel**, **OneNote**, **PowerPoint**, or **Word**. In this new document, select **Share** in the ribbon, select **Copy Link**, and copy the URL.

Reviewer 2: ```suggestion
1. Open [Office on the web](https://office.live.com/) or OneDrive. Using the **Create** option, make a document in **Excel**, **OneNote**, **PowerPoint**, or **Word**. In this new document, select **Share**, select **Copy Link**, and copy the URL.
```

1. In the command line starting at the root directory of your project, run the following command. Replace "{url}" with the URL that you copied.

    [!INCLUDE [npm start:web command syntax](../includes/start-web-sideload-instructions.md)]

1. The first time you use this method to sideload an add-in on the web, you'll see a dialog asking you to enable developer mode. Select the checkbox for **Enable Developer Mode now** and select **OK**.

> 1. You will see a second dialog box, asking if you wish to register an Office Add-in manifest from your computer. You should select **Yes**.

Reviewer 2: ```suggestion
1. You'll see a second dialog box, asking if you wish to register an Office Add-in manifest from your computer. Select **Yes**.
```

1. Your add-in is installed. If it has an add-in command, it should appear on either the ribbon or the context menu. If it's a task pane add-in without any add-in commands, the task pane should appear.

## Sideload an add-in on the web when using Visual Studio

> If you're using Visual Studio to develop your add-in, pressing F5 will automatically open an Office document in *desktop* Office, create a blank document, and sideload the add-in. When you want to sideload to *Office on the web*, the process to sideload is similar to manual sideloading to the web. The only difference is that you must update the value of the **SourceURL** element, and possibly other elements, in your manifest to include the full URL where the add-in is deployed.

Reviewer 2: ```suggestion
If you're using Visual Studio to develop your add-in, press **F5** to open an Office document in *desktop* Office, create a blank document, and sideload the add-in. When you want to sideload to *Office on the web*, the process to sideload is similar to manual sideloading to the web. The only difference is that you must update the value of the **SourceURL** element, and possibly other elements, in your manifest to include the full URL where the add-in is deployed.
```

> 1. In Visual Studio, show the **Properties** window by choosing **View** > **Properties Window**.

Reviewer 2: ```suggestion
1. In Visual Studio, choose **View** > **Properties Window**.
```

> 1. In the **Solution Explorer**, select the web project. This will display properties for the project in the **Properties** window.

Reviewer 2: ```suggestion
1. In the **Solution Explorer**, select the web project. This displays properties for the project in the **Properties** window.
```

1. In the Properties window, copy the **SSL URL**.

> 1. In the add-in project, open the manifest XML file. Be sure you are editing the source XML. For some project types, Visual Studio will open a visual view of the XML which won't work for the next step.

Reviewer 2: ```suggestion
1. In the add-in project, open the manifest XML file. Be sure you're editing the source XML. For some project types, Visual Studio will open a visual view of the XML which won't work for the next step.
```

1. Search and replace all instances of **~remoteAppUrl/** with the SSL URL you just copied. You'll see several replacements depending on the project type, and the new URLs will appear similar to `https://localhost:44300/Home.html`.

> 1. Save the XML file.

Reviewer 2: ```suggestion
1. **Save** the XML file.
```

1. In the **Solution Explorer**, open the context menu of the web project (for example, by right clicking on it) then choose **Debug** > **Start new instance**. This runs the web project without launching Office.

1. From Office on the web, sideload the add-in using steps described in [Manually sideload an add-in to Office on the web](#manually-sideload-an-add-in-to-office-on-the-web).

## Manually sideload an add-in to Office on the web

This method doesn't use the command line and can be accomplished using commands only within the host application (such as Excel).

1. Open [Office on the web](https://office.com/). Open a document in **Excel**, **OneNote**, **PowerPoint**, or  **Word**. 

> 1. On the **Insert** tab on the ribbon in the **Add-ins** section, choose **Office Add-ins**.

Reviewer 2: ```suggestion
1. On the **Insert** tab, in the **Add-ins** section, choose **Office Add-ins**.
```

    ![The upload add-in dialog with buttons for browse, upload, and cancel.](../images/upload-add-in.png)

> 1. Verify that your add-in is installed. For example, if it has an add-in command, it should appear on either the ribbon or the context menu. If it is a task pane add-in that has no add-in commands, the task pane should appear.

Reviewer: ```suggestion
1. Verify that your add-in is installed. For example, if it has an add-in command, it should appear on either the ribbon or the context menu. If it's a task pane add-in without add-in commands, the task pane should appear.
```

Author: "without" is ambiguous, but I fixed the contraction

> [!NOTE]
> To test your Office Add-in with Microsoft Edge with the original WebView (EdgeHTML), an additional configuration step is required. In a Windows Command Prompt, run the following line: `npx office-addin-dev-settings appcontainer EdgeWebView --loopback --yes`. This is not required when Office is using the Chromium-based Edge WebView2. For more information, see [Browsers used by Office Add-ins](../concepts/browsers-used-by-office-web-add-ins.md).

[!INCLUDE[Office settings tool not supported on Mac](../includes/tool-nonsupport-mac-note.md)]

## Sideload an add-in to Microsoft 365

1. Sign in to your Microsoft 365 account.

1. Open the App Launcher on the left end of the toolbar and select **Excel**, **PowerPoint**, or **Word**, and then create a new document.

> 1. Steps 3 - 6 are the same as in the section [Manually sideload an add-in to Office on the web](#manually-sideload-an-add-in-to-office-on-the-web).

Reviewer: I read this and kind of thought I needed to follow steps 3-6 of the "Manually sideload an add-ins to Office on the web" section.

If you accept my suggestion for that section, then this can say something like:
```suggestion
1. Follow steps 2 - 5 of the [Manually sideload an add-in to Office on the web](#manually-sideload-an-add-in-to-office-on-the-web) section.
```

> 1. Steps 3 - 6 are the same as in the section [Manually sideload an add-in to Office on the web](#manually-sideload-an-add-in-to-office-on-the-web)..

Reviewer: ```suggestion
1. Steps 3 - 6 are the same as in the section [Manually sideload an add-in to Office on the web](#manually-sideload-an-add-in-to-office-on-the-web).
```

## Remove a sideloaded add-in

> You can remove an add-in sideloaded to Office on the web by clearing your browser's cache. If you make changes to your add-in's manifest (for example, update file names of icons or text of add-in commands), you may need to clear your browser's cache and then re-sideload the add-in using the updated manifest. Doing so will allow Office on the web to render the add-in as it's described by the updated manifest.

Reviewer: ```suggestion
You can remove an add-in sideloaded to Office on the web by clearing your browser's cache. If you make changes to your add-in's manifest (for example, update file names of icons or text of add-in commands), you may need to clear your browser's cache and then re-sideload the add-in using the updated manifest. Doing so allows Office on the web to render the add-in as it's described by the updated manifest.
```

    ![The upload add-in dialog with buttons for browse, upload, and cancel.](../images/upload-add-in.png)

1. Verify that your add-in is installed. For example, if it has an add-in command, it should appear on either the ribbon or the context menu. If it's a task pane add-in that has no add-in commands, the task pane should appear.

> [!NOTE]
> To test your Office Add-in with Microsoft Edge with the original WebView (EdgeHTML), an additional configuration step is required. In a Windows Command Prompt, run the following line: `npx office-addin-dev-settings appcontainer EdgeWebView --loopback --yes`. This is not required when Office is using the Chromium-based Edge WebView2. For more information, see [Browsers used by Office Add-ins](../concepts/browsers-used-by-office-web-add-ins.md).

[!INCLUDE[Office settings tool not supported on Mac](../includes/tool-nonsupport-mac-note.md)]

## Sideload an add-in to Microsoft 365

1. Sign in to your Microsoft 365 account.

1. Open the App Launcher on the left end of the toolbar and select **Excel**, **OneNote**, **PowerPoint**, or **Word**, and then create a new document.

> 1. On the **Insert** tab on the ribbon select the **Add-ins** button.

Reviewer 2: ```suggestion
1. On the **Insert** tab, select the **Add-ins** button.
```

1. Follow steps 3 - 5 of the section [Manually sideload an add-in to Office on the web](#manually-sideload-an-add-in-to-office-on-the-web).

## Remove a sideloaded add-in

> You can remove an add-in sideloaded to Office on the web by clearing your browser's cache. If you make changes to your add-in's manifest (for example, update file names of icons or text of add-in commands), you may need to clear your browser's cache and then re-sideload the add-in using the updated manifest. Doing so allows Office on the web to render the add-in as it's described by the updated manifest.

Reviewer 2: ```suggestion
To remove an add-in sideloaded to Office on the web, simply clear your browser's cache. If you make changes to your add-in's manifest (for example, update file names of icons or text of add-in commands), you may need to clear your browser's cache and then re-sideload the add-in using the updated manifest. Doing so allows Office on the web to render the add-in as it's described by the updated manifest.
```
